build-system/eslint-rules/core-dom-jsx.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @fileoverview Lints JSX features not supported by core/dom/jsx
+ */
+
+module.exports = function (context) {
+  let isCoreDomJsx = false;
+  return {
+    Program() {
+      isCoreDomJsx = false;
+    },
+    ImportNamespaceSpecifier(node) {
+      if (node.parent.source.value.endsWith('core/dom/jsx')) {
+        isCoreDomJsx = true;
+      }
+    },
+    JSXFragment(node) {
+      if (!isCoreDomJsx) {
+        return;
+      }
+      context.report({
+        node,
+        message:
+          'Fragments are not supported. Change into an array of elements, or wrap in a root element.',
+      });
+    },
+    JSXOpeningElement(node) {
+      if (!isCoreDomJsx) {
+        return;
+      }
+      if (node.name.name === 'foreignObject') {
+        context.report({
+          node: node.name,
+          message: `<${node.name.name}> is not supported.`,
+        });
+      }
+    },
+    JSXAttribute(node) {
+      if (!isCoreDomJsx) {
+        return;
+      }
+      if (node.name.name === 'dangerouslySetInnerHTML') {
+        context.report({
+          node: node.name,
+          message: `\`<${node.name.name}>\` is not supported.`,
+        });
+        return;
+      }
+      if (
+        !node.value ||
+        node.value.type === 'Literal' ||
+        node.value.expression?.type === 'Literal' ||
+        node.value.expression?.type === 'TemplateLiteral'
```

Comment:
Is the `expression?.type` optional chain because of JSX nodes as direct values of attributes? I'm surprised you caught that.

Reply:
Wow, I did not know this was possible! I used an optional chain out of habit only.

<img src="https://user-images.githubusercontent.com/254946/141218379-c3b5ee1b-6762-448f-a7ba-717f02affb12.png" width=585>